sw/airborne/modules/datalink/gec_dl.c

```diff
@@ -0,0 +1,382 @@
+/*
+ * Copyright (C) 2017 Michal Podhradsky <michal.podhradsky@aggiemail.usu.edu>
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/** \file modules/datalink/spprz_dl.c
+ *  \brief Datalink using Galois Embedded Crypto
+ */
+
+#include "gec_dl.h"
+#include "modules/datalink/gec/gec.h"
+#include "subsystems/datalink/datalink.h"
+#include "pprzlink/messages.h"
+#include <string.h> // for memset()
+
+#if PPRZLINK_DEFAULT_VER != 2
+#error "Secure link is only for Pprzlink v 2.0"
+#endif
+
+struct gec_transport gec_tp;
+
+#if PERIODIC_TELEMETRY
+#include "subsystems/datalink/telemetry.h"
+
+static void send_secure_link_info(struct transport_tx *trans, struct link_device *dev)
+{
+#if PPRZLINK_DEFAULT_VER == 2
+  struct pprzlink_msg msg;
+  msg.trans = trans;
+  msg.dev = dev;
+  msg.sender_id = AC_ID;
+  msg.receiver_id = PPRZLINK_MSG_BROADCAST;
```

Comment:
This is only needed if you want to broadcast the message to all aircraft around you and not only the GCS. If only the GCS should receive the message, using `pprz_msg_send_SECURE_LINK_STATUS` is fine and will work for both protocol.

Reply:
Fixed.